stl/inc/flat_map

```diff
@@ -960,210 +886,530 @@ private:
+    template <class _KeyTy>
+    iterator _Upper_bound(const _KeyTy& _Key_val)
+        requires same_as<_KeyTy, key_type> || _Is_transparent_v<key_compare>
     {
-        auto _Key_it = _STD upper_bound(_Data.keys.cbegin(), _Data.keys.cend(), _X, _Key_compare);
+        auto _Key_it = _STD upper_bound(_Data.keys.cbegin(), _Data.keys.cend(), _Key_val, _Key_compare);
         auto _Dist   = _STD distance(_Data.keys.cbegin(), _Key_it);
         auto _Val_it = _Data.values.begin() + _Dist;
         return iterator{_STD move(_Key_it), _STD move(_Val_it)};
     }
 
-    template <class _K>
-    const_iterator _Upper_bound(const _K& _X) const
-        requires same_as<_K, key_type> || _Is_transparent_v<key_compare>
+    template <class _KeyTy>
+    const_iterator _Upper_bound(const _KeyTy& _Key_val) const
+        requires same_as<_KeyTy, key_type> || _Is_transparent_v<key_compare>
     {
-        auto _Key_it = _STD upper_bound(_Data.keys.cbegin(), _Data.keys.cend(), _X, _Key_compare);
+        auto _Key_it = _STD upper_bound(_Data.keys.cbegin(), _Data.keys.cend(), _Key_val, _Key_compare);
         auto _Dist   = _STD distance(_Data.keys.cbegin(), _Key_it);
         auto _Val_it = _Data.values.cbegin() + _Dist;
         return const_iterator{_STD move(_Key_it), _STD move(_Val_it)};
     }
 
-    template <class _K>
-    pair<iterator, iterator> _Equal_range(const _K& _X)
-        requires same_as<_K, key_type> || _Is_transparent_v<key_compare>
+    template <class _KeyTy>
+    pair<iterator, iterator> _Equal_range(const _KeyTy& _Key_val)
+        requires same_as<_KeyTy, key_type> || _Is_transparent_v<key_compare>
     {
-        return {lower_bound(_X), upper_bound(_X)};
+        return {lower_bound(_Key_val), upper_bound(_Key_val)};
     }
 
-    template <class _K>
-    pair<const_iterator, const_iterator> _Equal_range(const _K& _X) const
-        requires same_as<_K, key_type> || _Is_transparent_v<key_compare>
+    template <class _KeyTy>
+    pair<const_iterator, const_iterator> _Equal_range(const _KeyTy& _Key_val) const
+        requires same_as<_KeyTy, key_type> || _Is_transparent_v<key_compare>
     {
-        return {lower_bound(_X), upper_bound(_X)};
+        return {lower_bound(_Key_val), upper_bound(_Key_val)};
     }
 };
 
 template <class _Key, class _Mapped, class _Compare, class _KeyContainer, class _MappedContainer>
-class flat_map : public _Flat_Map_Base<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer, false,
+class flat_map : public _Flat_map_base<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer, false,
                      flat_map<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer>> {
 private:
-    using _MyBase = _Flat_Map_Base<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer, false,
+    using _MyBase = _Flat_map_base<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer, false,
         flat_map<_Key, _Mapped, _Compare, _KeyContainer, _MappedContainer>>;

```

Comment:
Pre-existing: Using the injected-class-name for `flat_map` here would be significantly shorter, although it would break the symmetry with the actual base class mention above.